### PR TITLE
Add TextMate External Editor

### PR DESCRIPTION
**Pull request by caiofbpa**

### CODE REVIEW

File: app/src/lib/editors/darwin.ts [hunk 1]
```diff
@@ -113,7 +121,7 @@ async function findApplication(editor: ExternalEditor): Promise<string | null> {
  */
 export async function getAvailableEditors(): Promise<
   ReadonlyArray<IFoundEditor<ExternalEditor>>
-> {
+  > {
```
Comment (j-f1) [inline app/src/lib/editors/darwin.ts]: Can you remove the two spaces here?

Comment (caiofbpa) [reply app/src/lib/editors/darwin.ts]: Sure, I have no idea how those spaces wildly appeared there.

**Comment by caiofbpa** [comment]
Is there anything else to be done before merging this?

**Comment by NiklasBr** [comment]
Maybe also add TextMate to [docs/technical/editor-integration.md](https://github.com/desktop/desktop/blob/master/docs/technical/editor-integration.md)?

**Comment by caiofbpa** [comment]
@NiklasBr, done! Anything else?

**Comment by iAmWillShepherd** [comment]
@caiofbpa just need you to fix conflicts before I can merge.

**Comment by caiofbpa** [comment]
Finally done. 😄

**Pull request merged**